extensions/keycloak-sms-authenticator/src/main/java/com/inventage/keycloak/sms/credential/SmsCredentialProvider.java

```diff
@@ -29,30 +35,65 @@ public CredentialModel createCredential(RealmModel realm, UserModel user, SmsCre
             credentialModel.setCreatedDate(Time.currentTimeMillis());
         }
 
-        return allowMultipleCredentials() ? addCredential(user, credentialModel) : updateCredential(realm, user, credentialModel);
+        return allowMultipleCredentials() ? addCredential(realm, user, credentialModel) : updateCredential(realm, user, credentialModel);
     }
 
     private boolean allowMultipleCredentials() {
         return true;
     }
 
-    private CredentialModel addCredential(UserModel user, SmsCredentialModel credentialModel) {
-        return user.credentialManager().createStoredCredential(credentialModel);
+    private CredentialModel addCredential(RealmModel realm, UserModel user, SmsCredentialModel newCredential) {
+        if (samePhoneNumberAlreadyConfigured(user, newCredential)) {
+            LOGGER.debugf("addCredential: User already has a credential with same phone number. So old credential will be updated.");
+            final CredentialModel oldCredential = getSmsCredentialModel(realm, user);
+            return updateAvailableCredential(user, oldCredential, newCredential);
+        }
+
+        LOGGER.debugf("addCredential: User already has not a credential with same phone number. So new credential will be added.");
```

Comment:
I think it is a bit redundant saying that the user does not have a credential with the same phone number. I think just saying "A new SMS credential will be created" or something like that? Unless you want to keep the information that a new credential will be created **_because_** the user does not have one already with the same number then I would rewrite the sentence a little bit, like so.

```suggestion
        LOGGER.debugf("addCredential: User does not have a credential with the same phone number. So new credential will be added.");
```